pkg/flowaggregator/flowaggregator.go

```diff
@@ -550,3 +565,53 @@ func (fa *flowAggregator) fillK8sMetadata(key ipfixintermediate.FlowKey, record
 		}
 	}
 }
+
+func (fa *flowAggregator) fetchPodLabels(podAddress string) string {
+	pods, err := fa.podInformer.Informer().GetIndexer().ByIndex(podInfoIndex, podAddress)
+	if err == nil && len(pods) > 0 {
+		pod, ok := pods[0].(*corev1.Pod)
+		if !ok {
+			klog.Warningf("Invalid Pod obj in cache")
+		}
+		labelsJSON, err := json.Marshal(pod.GetLabels())
+		if err == nil {
+			return string(labelsJSON)
+		} else {
+			klog.Warningf("JSON encoding of Pod labels failed: %v", err)
+		}
+	} else if err != nil {
+		klog.Warning(err)
+	}
+	return ""
+}
+
+func (fa *flowAggregator) fillPodLabels(key ipfixintermediate.FlowKey, record ipfixentities.Record) {
+	if fa.podInformer == nil {
```

Comment:
Why we need a check here? It seems the informer is always set when initialized. If this is for testing only, we can just provide a fake informer to get rid of it?

Reply:
I agress this check is not needed, removed.